tests/failpoints/test_engine.rs

```diff
@@ -450,7 +450,75 @@ fn test_tail_corruption() {
         let engine = Engine::open_with_file_system(cfg.clone(), fs.clone()).unwrap();
         append(&engine, rid, 1, 5, Some(&data));
         drop(engine);
-        assert!(Engine::open_with_file_system(cfg, fs).is_err());
+        assert!(Engine::open_with_file_system(cfg, fs.clone()).is_err());
+    }
+    // Version::V1 in header owns abnormal DataLayout.
+    {
+        let _f = FailGuard::new("log_file_header::force_abnormal_data_layout", "return");
+        let dir = tempfile::Builder::new()
+            .prefix("test_tail_corruption_4")
+            .tempdir()
+            .unwrap();
+        let cfg_err = Config {
+            dir: dir.path().to_str().unwrap().to_owned(),
+            recovery_mode: RecoveryMode::AbsoluteConsistency,
```

Comment:
Shouldn't use this mode. The test checks whether the code correctly detects tail corruption, by entering a condition of tail corruption (and engine opens successfully), and entering a condition of non tail corruption (and engine fails to open). But when you set to `AbsoluteConsistency` mode, engine will always fail in both cases, so it's hard to reason if the engine is behaving correctly.